src/test/java/org/sagebionetworks/bridge/validators/AdherenceReportSearchValidatorTest.java

```diff
@@ -0,0 +1,139 @@
+        search.setPageSize(300);
+        
+        assertValidatorMessage(validator, search, "pageSize", PAGE_SIZE_ERROR);
+    }
+
+    @Test
+    public void tooManyLabelOptions() { 
+        List<String> labels = new ArrayList<>();
+        for (int i=0; i < 300; i++) {
+            labels.add("abc");
+        }
+        AdherenceReportSearch search = new AdherenceReportSearch();
+        search.setLabelFilters(labels);
+        
+        assertValidatorMessage(validator, search, "labelFilters", LABEL_FILTER_COUNT_ERROR);
+    }
+
+    @Test
+    public void nullLabelOption() { 
+        List<String> labels = new ArrayList<>();
+        labels.add(null);
+        AdherenceReportSearch search = new AdherenceReportSearch();
+        search.setLabelFilters(labels);
+        
+        assertValidatorMessage(validator, search, "labelFilters[0]", CANNOT_BE_BLANK);
+    }
+
+    @Test
+    public void blankLabelOption() { 
+        List<String> labels = new ArrayList<>();
+        labels.add(" ");
+        AdherenceReportSearch search = new AdherenceReportSearch();
+        search.setLabelFilters(labels);
+        
+        assertValidatorMessage(validator, search, "labelFilters[0]", CANNOT_BE_BLANK);
+    }
+    
+    @Test
+    public void negativeLabelTooLong() { 
+        List<String> labels = new ArrayList<>();
+        labels.add(StringUtils.repeat("A", 300));
+        AdherenceReportSearch search = new AdherenceReportSearch();
+        search.setLabelFilters(labels);
+        
+        assertValidatorMessage(validator, search, "labelFilters[0]", LABEL_FILTER_LENGTH_ERROR);
+    }
+
+    @Test
+    public void adherenceMinTooLow() { 
+        AdherenceReportSearch search = new AdherenceReportSearch();
+        search.setAdherenceMin(-1);
+        
+        assertValidatorMessage(validator, search, "adherenceMin", ADHERENCE_RANGE_ERROR);
+    }
+
+    @Test
+    public void adherenceMinTooHigh() { 
+        AdherenceReportSearch search = new AdherenceReportSearch();
+        search.setAdherenceMin(101);
+        
+        assertValidatorMessage(validator, search, "adherenceMin", ADHERENCE_RANGE_ERROR);
+    }
+
+    @Test
+    public void adherenceMaxTooLow() { 
+        AdherenceReportSearch search = new AdherenceReportSearch();
+        search.setAdherenceMax(-1);
+        
+        assertValidatorMessage(validator, search, "adherenceMax", ADHERENCE_RANGE_ERROR);
+    }
+
+    @Test
+    public void adherenceMaxTooHigh() { 
+        AdherenceReportSearch search = new AdherenceReportSearch();
+        search.setAdherenceMax(101);
+        
+        assertValidatorMessage(validator, search, "adherenceMax", ADHERENCE_RANGE_ERROR);
+    }
+
+    @Test
+    public void adherenceMaxHigherThanMin() { 
+        AdherenceReportSearch search = new AdherenceReportSearch();
+        search.setAdherenceMax(101);
+        
+        assertValidatorMessage(validator, search, "adherenceMax", ADHERENCE_RANGE_ERROR);
+    }
+}
```

Comment:
This needs an adherenceMin set above the max and an updated error message.

Reply:
Yep. Changed test and turned the error message into a constant.